Exiled.Events/Patches/Events/Player/ActivatingWarheadPanel.cs

```diff
@@ -7,72 +7,151 @@
+            //
+            // Handlers.Player.OnActivatingWarheadPanel(ev);
+            //
+            // ListPool<string>.Shared.Return(list);
+            //
+            // if (ev.IsAllowed)
+            //   goto insideIfLabel;
+            //
+            // return;
+            newInstructions.InsertRange(index, new[]
             {
-                API.Features.Log.Error($"{typeof(ActivatingWarheadPanel).FullName}.{nameof(Prefix)}:\n{exception}");
+                // if (this._sr.BypassMode)
+                //   goto continueLabel
+                new CodeInstruction(OpCodes.Ldarg_0),
+                new CodeInstruction(OpCodes.Ldfld, Field(typeof(PlayerInteract), nameof(PlayerInteract._sr))),
+                new CodeInstruction(OpCodes.Ldfld, Field(typeof(ServerRoles), nameof(ServerRoles.BypassMode))),
+                new CodeInstruction(OpCodes.Brtrue_S, continueLabel),
+
+                // if (itemByID == null)
+                //   return;
+                new CodeInstruction(OpCodes.Ldloc_1),
+                new CodeInstruction(OpCodes.Brfalse_S, returnLabel),
+
+                // var list = ListPool<string>.Shared.Rent()
+                new CodeInstruction(OpCodes.Ldsfld, Field(typeof(ListPool<string>), nameof(ListPool<string>.Shared))).WithLabels(continueLabel),
+                new CodeInstruction(OpCodes.Callvirt, Method(typeof(ListPool<string>), nameof(ListPool<string>.Rent), new Type[] { })),
+                new CodeInstruction(OpCodes.Dup),
+                new CodeInstruction(OpCodes.Stloc_S, list.LocalIndex),
+
+                // list.Add("CONT_LVL_3)
+                new CodeInstruction(OpCodes.Ldstr, permission),
+                new CodeInstruction(OpCodes.Call, Method(typeof(List<string>), nameof(List<string>.Add))),
+
+                // Player.Get(this.gameObject)
+                new CodeInstruction(OpCodes.Ldarg_0),
+                new CodeInstruction(OpCodes.Call, PropertyGetter(typeof(Component), nameof(Component.gameObject))),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(GameObject) })),
+
+                // list
+                new CodeInstruction(OpCodes.Ldloc_S, list.LocalIndex),
+
+                // if (this._sr.BypassMode)
+                //   goto checkedBypassLabel;
+                new CodeInstruction(OpCodes.Ldarg_0),
+                new CodeInstruction(OpCodes.Ldfld, Field(typeof(PlayerInteract), nameof(PlayerInteract._sr))),
+                new CodeInstruction(OpCodes.Ldfld, Field(typeof(ServerRoles), nameof(ServerRoles.BypassMode))),
+                new CodeInstruction(OpCodes.Brtrue_S, checkedBypassLabel),
+
+                // itemByID.permissions.Contains("CONT_LVL_3")
+                //   goto checkedContainsLabel
+                new CodeInstruction(OpCodes.Ldloc_1),
+                new CodeInstruction(OpCodes.Ldfld, Field(typeof(global::Item), nameof(global::Item.permissions))),
```

Comment:
Is `global::` necessary in this context?

Reply:
Yes, since Item without `global::` would generate conflicts.